src/main/kotlin/no/nav/klage/oppgave/service/OppgaveService.kt

```diff
@@ -139,27 +135,53 @@ class OppgaveService(
     private fun Oppgave.getFnrForBruker() = identer?.find { i -> i.gruppe == FOLKEREGISTERIDENT }?.ident
 
     fun assignRandomHjemler(): List<OppgaveView> {
-        val oppgaver = getOppgaver().map {
-            oppgaveClient.endreHjemmel(it.id, hjemler.random())
-        }
+        val oppgaver = oppgaveClient.getOppgaver().oppgaver
+            .map { it.toEndreOppgave() }
+            .map {
+                it.setHjemmel(hjemler.random())
+                oppgaveClient.putOppgave(it.id, it)
+            }
         val brukere = getBrukere(getFnr(oppgaver))
         return oppgaver.map {
             toView(it, brukere, emptyMap())
         }
     }
 
     fun setHjemmel(oppgaveId: Int, hjemmel: String): OppgaveView {
-        val oppgave = oppgaveClient.endreHjemmel(oppgaveId, hjemmel)
+        var oppgave = oppgaveClient.getOppgave(oppgaveId).toEndreOppgave()
+        oppgave.setHjemmel(hjemmel)
+        return updateAndReturn(oppgaveId, oppgave)
+    }
+
+    private fun EndreOppgave.setHjemmel(hjemmel: String) {
+        if (metadata == null) {
```

Comment:
Lurer på om metadata kunne vært `val` og default mutableMap isf `null`. Vet ikke om det ville fått noen negative konsekvenser her. Ofte tenker jeg at collections i data-klasser aldri skal kunne være `null`. Koden blir gjerne mer letthåndterlig og lettlest da.

Reply:
Morsomt at du skriver det, for jeg skrev koden sånn først, men så endret jeg tilbake. Jeg var ikke 100% sikker på hvordan Oppgave-systemet ville takle et tomt mutableMap, så da tenkte jeg det var greit å la være. Men jeg er egentlig helt enig med deg i alt du skriver, så jeg kan gjerne endre tilbake.. :)

Reply:
Jeg prøver å gjøre om, men tar det som en egen PR.